src/Game/Enemy/FireBall.cpp

```diff
@@ -0,0 +1,201 @@
+    if (MR::isSensorPlayer(pReceiver) && MR::sendMsgEnemyAttackFire(pReceiver, pSender)) {
+        kill();
+        return;
+    }
+    if (MR::isSensorEnemy(pReceiver) && pReceiver->mHost != this->_8C && MR::sendMsgEnemyAttackFire(pReceiver, pSender) ) {
+        kill();
+    }
+}
+
+bool FireBall::receiveMsgPlayerAttack(u32 msg, HitSensor *pSender, HitSensor *pReceiver) {
+    if (MR::isMsgJetTurtleAttack(msg)) {
+        kill();
+        return false;
+    }
+    if (MR::isMsgLockOnStarPieceShoot(msg)) {
+        return true;
+    }
+
+    if (MR::isMsgStarPieceAttack(msg)) {
+        kill();
+        return true;
+    }
+    return false;
+}
+
+HitSensor* FireBall::isBindedAny() const {
+    if (MR::isBindedGround(this)) {
+        return MR::getGroundSensor(this);
+    }
+    if (MR::isBindedWall(this)) {
+        return MR::getWallSensor(this);
+    }
+    if (MR::isBindedRoof(this)) {
+        return MR::getRoofSensor(this);
+    }
+    return nullptr;
+}
+
+bool FireBall::tryToKill() {
+    HitSensor* bindedSensor  = isBindedAny();
+
+    if (bindedSensor) {
+        MR::sendMsgEnemyAttack(bindedSensor, getSensor("body"));
+        kill();
+        return true;
+    } else if (!MR::isNearPlayer(this, 5000.0f)) {
+        kill();
+        return true;
+    }
+
+    return false;
+}
+
+void FireBall::setVelocityToPlayer(f32 param1) {
+    TVec3f stack_14;
+    stack_14.scale(120.0f, _90);
+    stack_14.add(*MR::getPlayerPos());
+    TVec3f a1;
+    JMathInlineVEC::PSVECSubtract2(&stack_14, &mPosition, &a1);
+    MR::normalize(&a1);
+    mVelocity.scale(param1, a1);
+
+}
+
+void FireBall::calcReflectVelocity() {
+    MR::getStarPointerWorldVelocityDirection(&mVelocity, *MR::getStarPointerLastPointedPort(this));
+    if (MR::isNearZero(mVelocity, 0.001f)) {
+        mVelocity.set(_90);
+    } else {
+        MR::vecKillElement(mVelocity, _90, &mVelocity);
+        TVec3f v7;
+        v7.scale(MR::getRandom(0.0f, 1.0f), _90);
+        mVelocity.add(v7);
+        if (MR::isNearZero(mVelocity, 0.001f)) {
+            mVelocity.set(_90);
+        } else {
+            MR::normalize(&mVelocity);
+        }
+    }
+    mVelocity.mult(100.0f);
+}
+
+void FireBall::exeThrow() {
+    if (MR::isFirstStep(this)) {
+        MR::startBck(this, "Spin", nullptr);
+    }
+
+    if (MR::changeShowModelFlagSyncNearClipping(this, 200.0f)) {
+        MR::emitEffect(this, "FireBall");
+    } else {
+        MR::deleteEffect(this, "FireBall");
+    }
+
+    if (MR::isGreaterStep(this, 30) && MR::isStarPointerPointing2POnPressButton(this, "弱", true, false)) {
+        s32* StarPointerLastPointedPort = MR::getStarPointerLastPointedPort(this);
+        TVec2f PointerScreenVel = *MR::getStarPointerScreenVelocity(*StarPointerLastPointedPort);
+        if (30.0f < PointerScreenVel.length()) {
+            calcReflectVelocity();
+            setNerve(&NrvFireBall::FireBallNrvReflect::sInstance);
+            return;
+        }
+    }
+    if(tryToKill()) return;
```

Comment:
Correct the spacing between the `if` keyword and the opening parenthesis, and encapsulate the `return` statement in curly brackets like a multi-line block.